include/seqan3/core/algorithm/algorithm_range.hpp

```diff
@@ -18,93 +18,93 @@
 namespace seqan3
 {
 
-/*!\brief An input range over the alignment results generated by the underlying alignment executor.
- * \ingroup pairwise_alignment
+/*!\brief An input range over the algorithm results generated by the underlying algorithm executor.
+ * \ingroup algorithm
  * \implements std::ranges::input_range
  *
- * \tparam alignment_executor_type The type of the underlying alignment executor; must be of type
+ * \tparam algorithm_executor_type The type of the underlying algorithm executor; must be of type
  *                                 seqan3::detail::algorithm_executor_blocking.
  *
  * \details
  *
- * Provides a lazy input-range interface over the alignment results generated by the underlying alignment
- * executor. The alignments are computed in a lazy fashion, such that when incrementing the next alignment
- * result is fetched from the executor. The alignment result will be cached such that dereferencing the
+ * Provides a lazy input-range interface over the algorithm results generated by the underlying algorithm
+ * executor. The algorithms are computed in a lazy fashion, such that when incrementing the next algorithm
+ * result is fetched from the executor. The algorithm result will be cached such that dereferencing the
  * associated iterator is constant.
  *
  * \if DEV
  * Note that the required type is not enforced in order to test this class without adding the entire machinery for
  * the seqan3::detail::algorithm_executor_blocking.
  * \endif
  */
-template <typename alignment_executor_type>
-class alignment_range
+template <typename algorithm_executor_type>
+class algorithm_range
```

Comment:
```suggestion
class algorithm_result_range
```
The name `algorithm_range` rather suggests a range over several algorithms, no?

Reply:
Well, that's exactly the point. It is a range over algorithms meaning, that it has to execute the algorithms to get the result. The other name might suggest, that the algorithms have been already executed, and this is only a range over the results.

Reply:
Hmm I see your point, but `*alignment_range.begin()` will return an `alignment_result` or a `search_result` and not an algorithm? That the range is lazy is usually not reflected in the name.

Reply:
It does return a result that is true, but has to execute an algorithm for it. That is not trivial. I suggest to call it then `algorithm_result_generator_range` in order to reflect, that it generates something.

Reply:
Sorry for the late response.

If it is fine with you, I'd prefer `algorithm_result_generator_range` because it is more expressive. And since we usually do not explicitily state this name in snippets I think the rather long name is also fine. Thanks for discussing it :)